### PR TITLE
garble: init at 2020-01-07

### PR DESCRIPTION
**Pull request by DavHau**
###### Motivation for this change
Garble: Obfuscate Go code by wrapping the Go toolchain.

I have been using this tool to add some protection against analysis of binaries built via go.

###### Things doneObfuscate Go code by wrapping the Go toolchain. Requires Go 1.15 or later,

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112631` at de87a15a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/496c0a847d01a7a709858fd0a1b8df16">garble</a></li></li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2091,6 +2091,12 @@
     email = "christoph.senjak@googlemail.com";
     name = "Christoph-Simon Senjak";
   };
+  DavHau = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2094]: Please put this into a separate commit with the message ``maintainers: add davhau``.

File: pkgs/build-support/go/garble.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ buildGoModule
+, fetchFromGitHub
+, lib
+, ...
```
Comment (SuperSandro2000) [inline pkgs/build-support/go/garble.nix:4]:
```suggestion
```

File: pkgs/build-support/go/garble.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+}:
+buildGoModule rec {
+  pname = "garble";
+  version = "20200107";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/build-support/go/garble.nix:9]:
```suggestion
  version = "20200107";

  src = fetchFromGitHub {
```

File: pkgs/build-support/go/garble.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    owner = "burrowers";
+    repo = pname;
+    rev = "835f4aadf321521acf06aac4d5068473dc4b2ac1";
+    sha256 = "sha256-NodsVHRll2YZoxrhmniJvelQOStG82u3kJyc0t8OXD8=";
+  };
+  vendorSha256 = "sha256-x2fk2QmZDK2yjyfYdK7x+sQjvt7tuggmm8ieVjsNKek=";
```
Comment (SuperSandro2000) [inline pkgs/build-support/go/garble.nix:15]:
```suggestion
  };

  vendorSha256 = "sha256-x2fk2QmZDK2yjyfYdK7x+sQjvt7tuggmm8ieVjsNKek=";
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112631` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>garble: log <a href="https://termbin.com/tj72">https://termbin.com/tj72</a></li>
  </ul>
</details>
